pancancer_evaluation/data_models/tcga_data_model.py

```diff
@@ -0,0 +1,234 @@
+import sys
+import typing
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+from pancancer_evaluation.utilities.tcga_utilities import (
+    process_y_matrix,
+    align_matrices,
+    standardize_gene_features,
+    check_status
+)
+from pancancer_evaluation.exceptions import ResultsFileExistsError
+
+class TCGADataModel():
+    """
+    Class containing data necessary to run TCGA mutation prediction experiments.
+
+    Provides an interface to load and preprocess mutation and gene expression,
+    and to split it into train/test sets for each target gene.
+    """
+
+    def __init__(self,
+                 seed=cfg.default_seed,
+                 results_dir=cfg.results_dir,
+                 subset_mad_genes=-1,
+                 verbose=False,
+                 debug=False,
+                 test=False):
+        """
+        Initialize mutation prediction model/data
+
+        Arguments
+        ---------
+        seed (int): seed for random number generator
+        results_dir (str): where to write results files
+        subset_mad_genes (int): how many genes to keep (top by mean absolute deviation).
+                                -1 doesn't do any filtering (all genes will be kept).
+        verbose (bool): whether or not to write verbose output
+        debug (bool): if True, use a subset of expression data for quick debugging
+        test (bool): if True, don't save results to files
+        """
+        # save relevant parameters
+        np.random.seed(seed)
+        self.seed = seed
+        self.results_dir = results_dir
```

Comment:
Good point. I put it here because doing the file handling in the scripts that call this class results in a lot of boilerplate that's pretty similar between scripts, but I agree that the data model probably isn't the best place for it.

I'll move the repeated file handling code to a separate utilities file, and call that from my experiment scripts - that way it's separate from this data model, which makes more sense to me as well.